packages/salesforcedx-vscode-core/src/commands/commands.ts

```diff
@@ -145,6 +227,29 @@ export class SelectDirPath
     this.globKeyWord = globKeyWord;
   }
 
+  public abstract globDirs(srcPath: string, priorityKeyword?: string): string[];
+
+  public async gather(): Promise<
+    CancelResponse | ContinueResponse<{ outputdir: string }>
+  > {
+    const rootPath = vscode.workspace.rootPath;
+    let outputdir;
+    if (rootPath) {
+      outputdir = this.explorerDir
+        ? path.relative(rootPath, this.explorerDir)
+        : await vscode.window.showQuickPick(
+            this.globDirs(rootPath, this.globKeyWord),
+            <vscode.QuickPickOptions>{
+              placeHolder: nls.localize('parameter_gatherer_enter_dir_name')
+            }
+          );
+    }
+    return outputdir
+      ? { type: 'CONTINUE', data: { outputdir } }
+      : { type: 'CANCEL' };
+  }
+}
+export class SelectPrioritizedDirPath extends SelectDirPath {
```

Comment:
Lists the dirs and moves dirs containing the keyword as a subdirectory to the top of the list